deps/coro-fs.lua

```diff
@@ -130,7 +131,22 @@ function fs.readFile(path)
   if err then return nil, err end
   stat, err = fs.fstat(fd)
   if stat then
-    data, err = fs.read(fd, stat.size)
+    --special case files on virtual filesystem
+    if stat.size == 0 and stat.birthtime.sec == 0 and stat.birthtime.nsec == 0 then
+      -- handle magic files the kernel generates as requested.
+      -- hopefully the heuristic works everywhere
+      local buffs = {}
+      local offs = 0
+      local size = 1024 * 48
+      repeat
+        data, err = fs.read(fd, size, offs)
+        table.insert(buffs, data or "")
+      until err or #data < size
+      return table.concat(buffs), err
```

Comment:
This should return `nil, err` on error rather than `"", err`. Also this early return means that `uv.fs_close` is never called.

Should probably change this line to something like
```lua
if not err then
  data = table.concat(buffs)
end
```
instead of returning early.

Reply:
Done.